src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -2020,11 +2020,26 @@ internal Task<Compilation> GetCompilationAsync(Project project, CancellationToke
                 : SpecializedTasks.Default<Compilation>();
         }
 
+        /// <summary>
+        /// Return reference completeness for the given project.
+        /// </summary>
         internal Task<bool> HasCompleteReferencesAsync(Project project, CancellationToken cancellationToken)
         {
+            // return true when compilation is not supported, otherwise, things like diagnostic service won't work for F# or Typescript.
             return project.SupportsCompilation
                 ? this.GetCompilationTracker(project.Id).HasCompleteReferencesAsync(this, cancellationToken)
-                : SpecializedTasks.False;
+                : SpecializedTasks.True;
+        }
+
+        /// <summary>
+        /// Return reference completeness for the given project and all projects this references.
+        /// </summary>
+        internal Task<bool> HasDependentCompleteReferencesAsync(Project project, CancellationToken cancellationToken)
+        {
+            // return true when compilation is not supported, otherwise, things like diagnostic service won't work for F# or Typescript.
```

Comment:
I think it should return projectInfo.IsComplete when compilation is not supported.